components/editable-description.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import { useState } from "react";
+import { LucidePencil, CheckSquare, XSquare } from "lucide-react";
+import { useRouter } from "next/navigation";
+
+import { updateProject } from "@/lib/actions";
+import { DesignProject } from "@/lib/types";
+
+import { CardContent, CardHeader, CardDescription } from "@/components/ui/card";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from "@/components/ui/use-toast";
+import { Button } from "./ui/button";
+
+export default function EditableDescription({
+  project,
+}: {
+  project: DesignProject;
+}) {
+  const router = useRouter();
+
+  const [description, setDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [originalDescription, setOriginalDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [editing, setEditing] = useState(false);
+
+  async function clickUpdateProject() {
+    try {
+      const updates = { description: description };
+      await updateProject(project.id, updates);
+      setOriginalDescription(description);
```

Comment:
here I think we should use the returned value of updateProject

Reply:
fix all the issues